src/mscorlib/shared/System/Diagnostics/Tracing/IEventProvider.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32;
+
+#if ES_BUILD_STANDALONE
+namespace Microsoft.Diagnostics.Tracing
+#else
+namespace System.Diagnostics.Tracing
+#endif
+{
+    // Represents the interface between EventProvider and an external logging mechanism.
+    internal interface IEventProvider
+    {
+        // Register an event provider.
+        unsafe uint EventRegister(
+            ref Guid providerId,
+            UnsafeNativeMethods.ManifestEtw.EtwEnableCallback enableCallback,
+            void* callbackContext,
+            ref long registrationHandle);
+
+        // Unregister an event provider.
+        uint EventUnregister(long registrationHandle);
+
+        // Write an event.
+        unsafe int EventWriteTransferWrapper(
+            long registrationHandle,
+            ref EventDescriptor eventDescriptor,
+            Guid* activityId,
+            Guid* relatedActivityId,
+            int userDataCount,
+            EventProvider.EventData* userData);
+
+        // Get or set the per-thread activity ID.
+        int EventActivityIdControl(UnsafeNativeMethods.ManifestEtw.ActivityControl ControlCode, ref Guid ActivityId);
+
+        // Register an EventPipeEvent handle to this EventPipeEventProvider.
+        unsafe void AddEventHandle(Int64 keywords, uint eventID, uint eventVersion, uint level, bool needStack);
```

Comment:
an we renamed this to 'DefineEvent' or something like that?   It took me a while to understand the purpose.   I would also put the eventID first as it is the 'handle'.   

Is the plan to also pass information about the event parameters in this API?   It seems like we want to pass the name and 'signature description' (probably already serialized as it is easier to do this on the managed side.